microsoft/testsuites/core/timesync.py

```diff
@@ -1,9 +1,29 @@
 from pathlib import PurePosixPath
+from typing import List, Optional, Union
 
 from assertpy import assert_that
+from retry import retry
 
-from lisa import Environment, Node, TestCaseMetadata, TestSuite, TestSuiteMetadata
-from lisa.tools import Cat, Dmesg
+from lisa import Node, TestCaseMetadata, TestSuite, TestSuiteMetadata
+from lisa.tools import Cat, Dmesg, Lscpu
+from lisa.tools.lscpu import CpuType
+
+
+@retry(tries=150, delay=0.5)
+def _check_value(
+    node: Node,
+    path: str,
+    expected_value: Optional[Union[str, List[str]]],
+) -> str:
+    cat = node.tools[Cat]
+    result = cat.run(path, force_run=True)
+    if isinstance(expected_value, List):
+        if result.stdout in expected_value:
+            return result.stdout
+    else:
+        if expected_value == result.stdout:
+            return result.stdout
+    return result.stdout
```

Comment:
The logic looks complex with multiple exit entries. And it looks will return the result, even the expected value is not here. It's same as below logic, is it expected?

```python
result = cat.run(path, force_run=True)
return result.stdout
```

Reply:
sometimes we need some time to wait for the file content change, if there is not return in this method, we have to cat the file again after this method, then compare actual and expected results.

Reply:
The above code paths has no difference. All code paths just return the result.stdout.